ciao-controller/workload.go

```diff
@@ -122,6 +137,24 @@ func (c *controller) CreateWorkload(req types.Workload) (types.Workload, error)
 		return req, err
 	}
 
+	// check to see if this is a new tenant. If so, we need to add
+	// them to the datastore. We do not however want to launch a
+	// CNCI yet since this might be a request to upload a CNCI
+	// workload. Instead, we'll add the new tenant directly to the
+	// datastore. On first launch request, if the tenant doesn't yet
+	// have a cnci, it will get launched for them then.
+	tenant, err := c.ds.GetTenant(req.TenantID)
```

Comment:
I also spotted that "ciao-cli volume list" will start the CNCI. Is there another codepath we need to protect against there too?